src/compiler/commandLineParser.ts

```diff
@@ -435,6 +516,30 @@ namespace ts {
                 else {
                     fileNames.push(s);
                 }
+
+                function parseCustomTypeOption(opt: CommandLineOptionOfCustomType, value: string) {
+                    const map = opt.type;
+                    const key = (value || "").toLowerCase();
+                    if (hasProperty(map, key)) {
+                        return map[key];
+                    }
+                    else {
+                        const suggestedOption = getNamesOfCustomTypeFromCommandLineOptionsOfCustomType(opt);
+                        errors.push(createCompilerDiagnostic(opt.error, suggestedOption ? suggestedOption : undefined));
+                    }
+                }
+
+                function parseListTypeOption(opt: CommandLineOptionOfListType, value: string): (number | string)[] {
+                    const values = (value || "").split(",").filter(v => { return v != undefined; });

```

Comment:
if `value` is falsy - there is no point to do `split/filter` part knowing that it will yield nothing
```ts
const values = value && filter(value.split(", "), v => !!v)
if (!values) {
    return [];
}
```